src/Validation/DonationValidator.php

```diff
@@ -54,13 +57,61 @@ public function validate( Donation $donation ): ValidationResult {
 	}
 
 	public function needsModeration( Donation $donation ): bool {
-		// TODO: add TextPolicyValidator
 		$violations = [];
 
 		$violations[] = $this->amountPolicyValidator->validate( $donation->getAmount(), $donation->getInterval() );
 
+		$violations = array_merge(
+			$violations,
+			$this->getBadWordViolations( $donation )
+		);
+
 		$this->policyViolations = array_filter( $violations );
+
 		return !empty( $this->policyViolations );
 	}
 
+	public function getPolicyViolations() {
```

Comment:
A type hint here would be nice (in the docs `@return ConstraintViolation[]`, and `:array` in the code)